### PR TITLE
Add dev.overlayed.Overlayed

### PR DESCRIPTION
**Pull request by Hacksore**
### Overview
Overlayed is companion app for Discord that lets you have a movable overlay to see who is in the voice chat.

### Please confirm your submission meets all the criteria

- [x] Please describe the application briefly.
- [x] The domain used for the application ID is [controlled by the application developers][appid-domain] and the [application id guidelines][appid] are followed.
- [x] I have read the and followed all the [Submission and license requirements][reqs].
- [x] I have [built][build] and tested the submission locally ([see README.md](https://github.com/Hacksore/flathub/blob/overlayed/README.md)).
image
- [x] I am an author/developer/upstream contributor of the project. If not, I contacted upstream developers about this submission. **Link:**
  - [x] https://github.com/overlayeddev/overlayed/pull/172

[appid-domain]: https://docs.flathub.org/docs/for-app-authors/requirements#control-over-domain
[appid]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission

### CODE REVIEW

File: dev.overlayed.Overlayed.yaml [hunk 1]
```diff
@@ -0,0 +1,81 @@
+id: dev.overlayed.Overlayed
+
+runtime: org.gnome.Platform
+runtime-version: "46"
+sdk: org.gnome.Sdk//46
```
Comment (bbhtt) [inline dev.overlayed.Overlayed.yaml:5]:
```suggestion
sdk: org.gnome.Sdk
```

File: dev.overlayed.Overlayed.yaml [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+      - install -Dm644 dev.overlayed.Overlayed.metainfo.xml /app/share/metainfo/com.overlayed.Desktop.metainfo.xml
+
+    sources:
+      - type: file
+        url: https://raw.githubusercontent.com/overlayeddev/overlayed/14a6d2c8b022f7d44de4a57350e8f30ff35ef7dc/distribution/dev.overlayed.Overlayed.metainfo.xml
+        sha256: 35773ae27a47753852aac2ace795675cef222d5d9681e5cdd93d92b6d16831c5
+
+      - type: file
+        url: https://github.com/overlayeddev/overlayed/releases/download/v0.5.0/overlayed_0.5.0_amd64.deb
+        sha256: 9a5d7057b184b164b4865c150dcb86afe1780ad7b1f2f78bd79db1c7f786bd44
```
Comment (bbhtt) [inline dev.overlayed.Overlayed.yaml:71]:
```suggestion
        sha256: 9a5d7057b184b164b4865c150dcb86afe1780ad7b1f2f78bd79db1c7f786bd44
        dest-filename: overlayed.deb
```

File: flathub.json [hunk 3]
```diff
@@ -0,0 +1,3 @@
+{
+  "only-arches": ["x86_64"]
```
Comment (hfiguiere) [inline flathub.json:2]: why limiting this to a single architecture?

File: dev.overlayed.Overlayed.yaml [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+        cleanup:
+          - "*"
+
+  - shared-modules/libappindicator/libappindicator-gtk3-12.10.json
+
+  - name: overlayed
+    buildsystem: simple
+    build-commands:
+      - ar -x *.deb
+      - tar -xf data.tar.gz
```
Comment (bbhtt) [inline dev.overlayed.Overlayed.yaml:56]:
```suggestion
      - bsdtar -Oxf overlayed.deb 'data.tar.gz' | bsdtar -xf -
```

File: dev.overlayed.Overlayed.yaml [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+      - mkdir -p /app/bin
+      - mkdir -p /app/share/applications
+      - install -Dm755 usr/bin/* /app/bin
+      - install -Dm644 usr/share/icons/hicolor/512x512/apps/overlayed.png /app/share/icons/hicolor/512x512/apps/dev.overlayed.Overlayed.png
+      - install -Dm644 usr/share/applications/overlayed.desktop /app/share/applications/dev.overlayed.Overlayed.desktop
+      - install -Dm644 dev.overlayed.Overlayed.metainfo.xml /app/share/metainfo/com.overlayed.Desktop.metainfo.xml
```
Comment (bbhtt) [inline dev.overlayed.Overlayed.yaml:62]:
```suggestion
      - install -Dm644 dev.overlayed.Overlayed.metainfo.xml /app/share/metainfo/dev.overlayed.Overlayed.metainfo.xml
```

This is why it is better to use `${FLATPAK_ID}` everywhere.

File: dev.overlayed.Overlayed.yaml (hunk 1 above)
Comment (Hacksore) [reply dev.overlayed.Overlayed.yaml:5]: I there a reason I have to omit this? If I do this it's going to bust my  compile cache of my linux test device.

File: flathub.json (hunk 3 above)
Comment (Hacksore) [reply flathub.json:2]: no support for arm as of now

File: dev.overlayed.Overlayed.yaml (hunk 1 above)
Comment (bbhtt) [reply dev.overlayed.Overlayed.yaml:5]: No need to build again, you don't need to specify the version twice.

**Comment by Hacksore** [comment]
@bbhtt can we please kick off a build?

**Comment by bbhtt** [comment]
bot, build dev.overlayed.Overlayed

Feel free to start builds on your own if you need.

**Comment by ahkohd** [comment]
superb!

File: dev.overlayed.Overlayed.yaml [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+
+  - shared-modules/libappindicator/libappindicator-gtk3-12.10.json
+
+  - name: overlayed
+    buildsystem: simple
+    build-commands:
+      - bsdtar -Oxf overlayed.deb 'data.tar.gz' | bsdtar -xf -
+      - mkdir -p /app/bin
+      - mkdir -p /app/share/applications
+      - mkdir -p /app/share/metainfo
```
Comment (hfiguiere) [inline dev.overlayed.Overlayed.yaml:58]:
```suggestion
```

`install -D` takes care of it

**Comment by hfiguiere** [comment]
The error is because the id in the appstream file is incorrect.

**Comment by bbhtt** [comment]
YOu haven't changed the id to dev.overlayed in appstream file:

```
  <id>com.overlayed.Overlayed</id>
```

Launchable tag too.

**Comment by Hacksore** [comment]
bot, build dev.overlayed.Overlayed

**Comment by Hacksore** [comment]
bot, build dev.overlayed.Overlayed

**Comment by Hacksore** [comment]
bot, build dev.overlayed.Overlayed

**Comment by Hacksore** [comment]
bot, build dev.overlayed.Overlayed

**Comment by Hacksore** [comment]
bot, build dev.overlayed.Overlayed

**Comment by Hacksore** [comment]
bot, build dev.overlayed.Overlayed

**Comment by Hacksore** [comment]
@bbhtt / @hfiguiere Working on my end, let me know if there are any outstanding blockers but I think it's good to `/merge`?

image

**Pull request closed without merging**